resources/services/frontdoor/gen.hcl

```diff
@@ -0,0 +1,561 @@
+    }
+
+    column "routing_rule_properties_route_configuration" {
+      rename            = "route_configuration"
+      description       = "A reference to the routing configuration"
+      type              = "json"
+      generate_resolver = true
+    }
+
+    column "routing_rule_properties_rules_engine_id" {
+      rename      = "rules_engine_id"
+      description = "ID of a specific Rules Engine Configuration to apply to the route"
+    }
+
+    column "routing_rule_properties_web_application_firewall_policy_link_id" {
+      rename      = "web_application_firewall_policy_link_id"
+      description = "ID of the Web Application Firewall policy for each routing rule (if applicable)"
+    }
+
+    column "id" {
+      description = "Resource ID"
+    }
+
+    column "name" {
+      description = "Resource name"
+    }
+
+    column "type" {
+      description = "Resource type"
+    }
+
+    relation "azure" "frontdoor" "routing_rule_properties_frontend_endpoints" {
+      rename      = "frontend_endpoints"
+      description = "Frontend endpoints associated with the rule."
+
+      column "id" {
+        description = "Resource ID"
+      }
+    }
+  }
+
+  relation "azure" "frontdoor" "properties_load_balancing_settings" {
+    rename      = "load_balancing_settings"
+    description = "Load balancing settings for a backend pool associated with the Front Door instance"
+
+    column "load_balancing_settings_properties_resource_state" {
+      rename      = "resource_state"
+      description = "Resource status"
+    }
+
+    column "load_balancing_settings_properties_sample_size" {
+      rename      = "sample_size"
+      description = "The number of samples to consider for load balancing decisions"
+    }
+
+    column "load_balancing_settings_properties_successful_samples_required" {
+      rename      = "successful_samples_required"
+      description = "The number of samples within the sample period that must succeed"
+    }
+
+    column "load_balancing_settings_properties_additional_latency_milliseconds" {
+      rename      = "additional_latency_milliseconds"
+      description = "The additional latency in milliseconds for probes to fall into the lowest latency bucket"
+    }
+
+    column "id" {
+      description = "Resource ID"
+    }
+
+    column "name" {
+      description = "Resource name"
+    }
+
+    column "type" {
+      description = "Resource type"
+    }
+  }
+
+  relation "azure" "frontdoor" "properties_health_probe_settings" {
+    rename      = "health_probe_settings"
+    description = "Health probe settings for a backend pool associated with this Front Door instance"
+
+    column "health_probe_settings_properties_resource_state" {
+      rename      = "resource_state"
+      description = "Resource status"
+    }
+
+    column "health_probe_settings_properties_path" {
+      rename      = "path"
+      description = "The path to use for the health probe"
+    }
+
+    column "protocol" {
```

Comment:
Referring to a comment I made higher up, I think this should be `health_probe_settings_properties_protocol`. 

On a side note (not for this PR), I think we can make some improvements to cq-gen here:
 - It should warn you when you've defined a column that it didn't find in the source. I started implementing this but got stuck at some point; I might try again when I'm more familiar with the source.
 - It seems clunky that it requires you to specify these long column names when `health_probe_settings` is already in the table name--seems like that prefix shouldn't be necessary, but maybe I'm missing something?

Reply:
Yes, the 2nd option would be really great (basically, I think `skip_prefix` should work not only for embedded relations, but for straightforward stuff as well. Or the generator shouldn't add prefix by itself).